src/views/Reports/Reports.tsx

```diff
@@ -13,28 +17,57 @@ export interface ReportsPanelProps {
   containerRef: RefObject<HTMLDivElement>
 }
 
+const DownloadButton = () => {
+  const { dateRange } = useContext(ProfitAndLoss.Context)
+  const { auth, businessId, apiUrl } = useLayerContext()
+  return (
+    <Button
+      variant={ButtonVariant.secondary}
+      rightIcon={<DownloadCloud size={12} />}
+      onClick={async () => {
+        const month = (dateRange.startDate.getMonth() + 1).toString()
+        const year = dateRange.startDate.getFullYear().toString()
+        const getProfitAndLossCsv = Layer.getProfitAndLossCsv(
+          apiUrl,
+          auth.access_token,
+          {
+            params: {
+              businessId: businessId,
+              year: year,
+              month: month,
+            },
+          },
+        )
+        try {
+          const result = await getProfitAndLossCsv()
+          if (result?.data?.presignedUrl)
+            window.location.href = result.data.presignedUrl
+        } catch (e) {}
```

Comment:
Nervous about errors - lots of surface area for silent failure.  We have an error notification pattern, can you try seeing if there's anything useful for this case?

Reply:
Used the failed / retry button pattern from the bank transaction categorization approval button